x-pack/test/apm_api_integration/tests/metadata/event_metadata.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import expect from '@kbn/expect';
+import { get } from 'lodash';
+import { PROCESSOR_EVENT } from '../../../../plugins/apm/common/elasticsearch_fieldnames';
+import { ProcessorEvent } from '../../../../plugins/apm/common/processor_event';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { registry } from '../../common/registry';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const apmApiClient = getService('apmApiClient');
+  const esClient = getService('es');
+
+  async function getLastDocId(processorEvent: ProcessorEvent) {
+    const response = await esClient.search({
+      index: ['apm-*'],
+      body: {
+        query: {
+          bool: {
+            filter: [{ term: { [PROCESSOR_EVENT]: processorEvent } }],
+          },
+        },
+        size: 1,
+        sort: {
+          '@timestamp': 'desc',
+        },
+      },
+    });
+
+    return get(response.body.hits.hits[0]._source, `${processorEvent}.id`);
```

Comment:
Do we need `get`

```suggestion
    return response.body.hits.hits[0]._source[processorEvent].id;
```

or if optional:

```suggestion
    return response.body.hits.hits[0]._source?.[processorEvent]?.id;
```

Reply:
sure, ts will complain out of the box but I can add a type for _source.